src/Damage/ProbabilisticDamageModel.cc

```diff
@@ -140,7 +140,7 @@ initializeProblemStartup(DataBase<Dimension>& dataBase) {
       mInitialVolume(i) = mass(i)/rho(i) * mVolumeMultiplier;
       mVmin = std::min(mVmin, mInitialVolume(i));
       mVmax = std::max(mVmax, mInitialVolume(i));
-      Key seedi = mSeed;
```

Comment:
Why this change?  We use the Key typedef in case in the future (or on different architectures) we need to change this, so we can do it one place.  This change hardwires this choice only at this point.

Reply:
It seems that the Key typedef does not play well with boost::hash_combine on MacOS, looking at the commit log. I just tried reverting it on my Mac laptop and got the following error:


In file included from /Users/tsuji1/Code/ale3d/imports/spheral/build-osx_gcc-relwithdebinfo/src/Damage/weibullFlawDistributionOwenInst2d.cc:6:
/Users/tsuji1/Code/ale3d/imports/spheral/src/./Damage/weibullFlawDistributionOwen.cc: In instantiation of 'Spheral::Field<Dimension, std::vector<double> > Spheral::weibullFlawDistributionOwen(unsigned int, double, double, const FluidNodeList<Dimension>&, int, double, const Field<Dimension, int>&) [with Dimension = Dim<2>]':
/Users/tsuji1/Code/ale3d/imports/spheral/build-osx_gcc-relwithdebinfo/src/Damage/weibullFlawDistributionOwenInst2d.cc:15:108   required from here
/Users/tsuji1/Code/ale3d/imports/spheral/src/./Damage/weibullFlawDistributionOwen.cc:87:26: error: no matching function for call to 'hash_combine(Key&, long long unsigned int&)'
   87 |       boost::hash_combine(seedi, keyList(0,i));
      |       ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
In file included from /Users/tsuji1/Code/ale3d/lib/osx_gcc/relwithdebinfo/BOOST/1.74.0b/compiler-gcc12.2.0/cpp-14/include/boost/functional/hash.hpp:6,
                 from /Users/tsuji1/Code/ale3d/imports/spheral/src/./Mesh/MeshConstructionUtilities.hh:23,
                 from /Users/tsuji1/Code/ale3d/imports/spheral/src/./Mesh/MeshInline.hh:15,
                 from /Users/tsuji1/Code/ale3d/imports/spheral/src/./Mesh/Mesh.hh:252,
                 from /Users/tsuji1/Code/ale3d/imports/spheral/src/./NodeList/SmoothingScaleBase.hh:13,
                 from /Users/tsuji1/Code/ale3d/imports/spheral/src/./NodeList/FluidNodeListInline.hh:3,
                 from /Users/tsuji1/Code/ale3d/imports/spheral/src/./NodeList/FluidNodeList.hh:122,
                 from /Users/tsuji1/Code/ale3d/imports/spheral/src/./Field/NodeIteratorBaseInline.hh:4,
                 from /Users/tsuji1/Code/ale3d/imports/spheral/src/./Field/NodeIteratorBase.hh:84,
                 from /Users/tsuji1/Code/ale3d/imports/spheral/src/./Field/NodeIterators.hh:4,
                 from /Users/tsuji1/Code/ale3d/imports/spheral/src/./Field/FieldInline.hh:4,
                 from /Users/tsuji1/Code/ale3d/imports/spheral/src/./Field/Field.hh:256,
                 from /Users/tsuji1/Code/ale3d/imports/spheral/src/./NodeList/NodeListInline.hh:2,
                 from /Users/tsuji1/Code/ale3d/imports/spheral/src/./NodeList/NodeList.hh:229,
                 from /Users/tsuji1/Code/ale3d/imports/spheral/src/./Utilities/globalNodeIDsInline.hh:14,
                 from /Users/tsuji1/Code/ale3d/imports/spheral/src/./Utilities/globalNodeIDs.hh:63,
                 from /Users/tsuji1/Code/ale3d/imports/spheral/src/./Damage/weibullFlawDistributionOwen.cc:6:
/Users/tsuji1/Code/ale3d/lib/osx_gcc/relwithdebinfo/BOOST/1.74.0b/compiler-gcc12.2.0/cpp-14/include/boost/container_hash/hash.hpp:413:17: note: candidate: 'template<class T> void boost::hash_combine(std::size_t&, const T&)'
  413 |     inline void hash_combine(std::size_t& seed, T const& v)
      |                 ^~~~~~~~~~~~
/Users/tsuji1/Code/ale3d/lib/osx_gcc/relwithdebinfo/BOOST/1.74.0b/compiler-gcc12.2.0/cpp-14/include/boost/container_hash/hash.hpp:413:17: note:   template argument deduction/substitution failed:
/Users/tsuji1/Code/ale3d/imports/spheral/src/./Damage/weibullFlawDistributionOwen.cc:87:27: note:   cannot convert 'seedi' (type 'Key' {aka 'long long unsigned int'}) to type 'std::size_t&' {aka 'long unsigned int&'}
   87 |       boost::hash_combine(seedi, keyList(0,i));
      |                           ^~~~~

Reply:
Ah, in this case I'd suggest we make this change with an #Ifdef __APPLE__ to restrict to only that platform.  My worry is if in the future we change KeyTraits::Key to something incompatible with size_t in some way, this could turn into a subtle and hard to find bug.  We could either change just this line in this way, or change type overall type of KeyTraits::Key in src/Utilities/KeyTraits.hh.

Reply:
I've added the ifdefs in both ProbabilisticDamageModel.cc and weibullFlawDistributionOwen.cc to only use std::size_t on Mac.